src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java

```diff
@@ -384,8 +390,12 @@ private static boolean isOnlyUnicodeValidChars(String literal, Pattern pattern)
      * @return true if trail comment is present after ast token.
      */
     private boolean hasTrailComment(DetailAST ast) {
+        int lineNo = ast.getLineNo();
+
+        if (ast.getType() == TokenTypes.TEXT_BLOCK_CONTENT) {
+            lineNo = ast.getNextSibling().getLineNo();
```

Comment:
Might add a comment here noting that text blocks start with a newline

Reply:
Good idea, fixed.